### PR TITLE
AWS ECS - convert tfsec checks to use defsec logic

### LINKED ISSUE #1262: AWS ECS - convert tfsec checks to use defsec logic
Adapt AWS ECS resources to defsec's Go structs based on #1194 instructions

### PR DESCRIPTION
**Pull request by gabrielchl**
- removed `CheckTerraform` from all 3 ECS rules
- added adaption code for ECS
Resolves #1262

### CODE REVIEW

File: internal/app/tfsec/adapter/aws/ecs/adapt.go [hunk 1]
```diff
@@ -2,9 +2,91 @@ package ecs
...
+func adaptClusterSettings(resourceBlock block.Block) ecs.ClusterSettings {
+	if settingBlock := resourceBlock.GetBlock("setting"); settingBlock.IsNotNil() && settingBlock.GetAttribute("name").Equals("containerInsights") {
+		containerInsightsEnabled := settingBlock.GetAttribute("value").Equals("enabled")
+		return ecs.ClusterSettings{
+			ContainerInsightsEnabled: types.Bool(containerInsightsEnabled, settingBlock.Metadata()),
+		}
+	}
+
+	return ecs.ClusterSettings{
+		ContainerInsightsEnabled: types.Bool(false, resourceBlock.Metadata()),
```
Comment (owenrumney) [inline internal/app/tfsec/adapter/aws/ecs/adapt.go:42]: As this is not set, we treat it as the default value to show that it was unset in the check
```suggestion
		ContainerInsightsEnabled: types.BoolDefault(false, resourceBlock.Metadata()),
```

**Comment by owenrumney** [comment]
Thanks for this @gabrielchl  - only one small point about the default value setting otherwise looks great. Glad you were able to get to grips with the adapting process 👍

File: internal/app/tfsec/adapter/aws/ecs/adapt.go (hunk 1 above)
Comment (gabrielchl) [reply internal/app/tfsec/adapter/aws/ecs/adapt.go]: (added new commit to address this, also marked the "thing" at line 90 as default too)

**Comment by gabrielchl** [comment]
> Glad you were able to get to grips with the adapting process

yeah it was a bit tricky handling optional arguments, figuring out what metadata is, etc., but the other conversion PRs provided a lot of examples :)

**Review by owenrumney — APPROVED** [review]

**Pull request merged**